src/hotspot/cpu/x86/x86.ad

```diff
@@ -2168,6 +2168,19 @@ bool Matcher::vector_needs_partial_operations(Node* node, const TypeVect* vt) {
   return false;
 }
 
+// Return true if Vector::rearrange needs preparation of the shuffle argument
+bool Matcher::vector_needs_load_shuffle(BasicType elem_bt, int vlen) {
```

Comment:
I think the name needs to be more expressive. If I read it alone, then I would think that it is about all kinds of vectors ... and it is confusing because what is a "load shuffle"? Are we shuffling loads or loading shuffles?